server/src/internalClusterTest/java/org/opensearch/nodestats/NodeStatsIT.java

```diff
@@ -243,6 +259,205 @@ public void testNodeIndicesStatsDocStatusStatsCreateDeleteUpdate() {
+                ArrayList<String> level_arg = new ArrayList<>();
+                level_arg.add(testLevel.getRestName());
+
+                commonStatsFlags.setLevels(level_arg.toArray(new String[0]));
+            }
+            response = client().admin().cluster().prepareNodesStats().setIndices(commonStatsFlags).get();
+
+            response.getNodes().forEach(nodeStats -> {
+                try {
+                    XContentBuilder builder = XContentFactory.jsonBuilder();
+
+                    builder.startObject();
+                    builder = nodeStats.getIndices()
+                        .toXContent(
+                            builder,
+                            new ToXContent.MapParams(Collections.singletonMap("level", NodeIndicesStats.Fields.SHARDS.getRestName()))
+                        );
+                    builder.endObject();
+
+                    Map<String, Object> xContentMap = xContentBuilderToMap(builder);
+                    LinkedHashMap indicesStatsMap = (LinkedHashMap) xContentMap.get(NodeIndicesStats.Fields.INDICES.getRestName());
+                    LinkedHashMap indicesStats = (LinkedHashMap) indicesStatsMap.get(NodeIndicesStats.Fields.INDICES.getRestName());
+                    LinkedHashMap shardStats = (LinkedHashMap) indicesStatsMap.get(NodeIndicesStats.Fields.SHARDS.getRestName());
+                    switch (testLevel) {
+                        case SHARDS:
+                            assertFalse(shardStats.isEmpty());
+                            assertNull(indicesStats);
+                            break;
+                        case INDICES:
+                        case NODE:
+                        case NULL:
+                        case UNKNOWN:
+                            assertTrue(shardStats.isEmpty());
+                            assertNull(indicesStats);
+                            break;
+                    }
+
+                    builder = XContentFactory.jsonBuilder();
+
+                    builder.startObject();
+                    builder = nodeStats.getIndices()
+                        .toXContent(
+                            builder,
+                            new ToXContent.MapParams(Collections.singletonMap("level", NodeIndicesStats.Fields.INDICES.getRestName()))
+                        );
+                    builder.endObject();
+
+                    xContentMap = xContentBuilderToMap(builder);
+                    indicesStatsMap = (LinkedHashMap) xContentMap.get(NodeIndicesStats.Fields.INDICES.getRestName());
+                    indicesStats = (LinkedHashMap) indicesStatsMap.get(NodeIndicesStats.Fields.INDICES.getRestName());
+                    shardStats = (LinkedHashMap) indicesStatsMap.get(NodeIndicesStats.Fields.SHARDS.getRestName());
+                    switch (testLevel) {
+                        case SHARDS:
+                        case INDICES:
+                            assertNull(shardStats);
```

Comment:
Can we include deep verifications on the stats - if the aggregated value is as per the expectedStatus

Reply:
Added a test - `testNodeIndicesStatsWithAndWithoutAggregations` which calls the MockNodeIndicesStats with and without aggregations passed and compares the ToXContent. This is deep in the sense that it compares common stats for 
1.  CommonStatsFlags.Flag.Docs
2. CommonStatsFlags.Flag.Store,
3. CommonStatsFlags.Flag.Indexing,
4. CommonStatsFlags.Flag.Completion,
5. CommonStatsFlags.Flag.Flush,
6. CommonStatsFlags.Flag.FieldData,
7. CommonStatsFlags.Flag.QueryCache,
8. CommonStatsFlags.Flag.Segments